assets/js/datepicker.js

```diff
@@ -1,27 +1,37 @@
 /* global job_manager_datepicker */
-jQuery(document).ready(function($) {
+jQuery(document).on( 'ready', function() {
 	var datePickerOptions = {
 		altFormat  : 'yy-mm-dd',
 	};
+
 	if ( typeof job_manager_datepicker !== 'undefined' ) {
 		datePickerOptions.dateFormat = job_manager_datepicker.date_format;
 	}
 
-	$( 'input.job-manager-datepicker, input#_job_expires' ).each( function(){
-		var $hidden_input = $( '<input />', { type: 'hidden', name: $(this).attr( 'name' ) } ).insertAfter( $( this ) );
-		$(this).attr( 'name', $(this).attr( 'name' ) + '-datepicker' );
-		$(this).keyup( function() {
-			if ( '' === $(this).val() ) {
+	var initializeDatepicker = function ( targetInput ) {
+		var $hidden_input = jQuery( '<input />', { type: 'hidden', name: jQuery(targetInput).attr( 'name' ) } ).insertAfter( jQuery( targetInput ) );
+		jQuery(targetInput).attr( 'name', jQuery(targetInput).attr( 'name' ) + '-datepicker' );
+		jQuery(targetInput).keyup( function() {
+			if ( '' === jQuery(targetInput).val() ) {
 				$hidden_input.val( '' );
 			}
 		} );
-		$(this).datepicker( $.extend( {}, datePickerOptions, { altField: $hidden_input } ) );
-		if ( $(this).val() ) {
-			var dateParts = $(this).val().split("-");
+		jQuery(targetInput).datepicker( jQuery.extend( {}, datePickerOptions, { altField: $hidden_input } ) );
+		if ( jQuery(targetInput).val() ) {
+			var dateParts = jQuery(targetInput).val().split('-');
```

Comment:
hmm, I think maybe we should use `.attr("value")` to read the value instead of using `.val()`?

If I understand this correctly, the idea is to read the value that PHP added as the `value` of the field. However, apparently, the `.datepicker` above is changing the `.value` of the element (which is returned by jQuery's `.val()` ) to a formatted date, which doesn't use `-` anymore (at least not here on my PC).

If we use `.attr("value")`, it will run a classic `.getAttribute("value")`, which returns the original value in the HTML, which I think is what we want.

This solves the issue I found on the review here: 115-gh-Automattic/wpjm-addons#pullrequestreview-1634134670 

WDYT?

Reply:
See my comment above.